### PR TITLE
Add support for loading 32-bit libs if present

### PR DESCRIPTION
**Pull request by samuelraeburn**

### CODE REVIEW

File: rticonnextdds-connector.js [hunk 1]
```diff
@@ -34,47 +34,52 @@ class _ConnectorBinding {
...
-    // Obtain the name of the library that contains the Connector binding
-    if (os.arch() === 'x64') {
-      switch (os.platform()) {
+    // Obtain the name of the library that contains the Connector native libraries
+    if (os.arch() === 'arm64') {
+        if (os.platform() === 'linux') {
+            libDir = 'linux-arm64'
+            libName = 'librtiddsconnector.so'
+        } else {
+            throw new Error('This platform (' + os.platform() + ' ' + os.arch() + ') is not supported')
```
Comment (samuelraeburn) [inline rticonnextdds-connector.js:43]: Couldn't decide whether or not to throw an error here or try to continue.

File: rticonnextdds-connector.js [hunk 2]
```diff
@@ -34,47 +34,52 @@ class _ConnectorBinding {
...
+            libDir = 'linux-arm'
+            libName = 'librtiddsconnector.so'
+        } else {
+            throw new Error('This platform (' + os.platform() + ' ' + os.arch() + ') is not supported')
+        }
+    } else {
+        // Note that we are intentionally not checking if os.arch() is x64.
+        // This allows somebody with access to 32-bit libraries to replace them
+        // in the corresponding x64 directory and we will try to load them.
+        // This behaviour is not supported.
```
Comment (alexcamposruiz) [inline rticonnextdds-connector.js:56]:
```suggestion
        // This behaviour is not officially supported.
```

File: rticonnextdds-connector.js (hunk 1 above)
Comment (alexcamposruiz) [reply rticonnextdds-connector.js:43]: This is fine. What do you do in py?

**Review by alexcamposruiz — APPROVED** [review]
Only a question and a minor comment edit.

File: rticonnextdds-connector.js (hunk 1 above)
Comment (samuelraeburn) [reply rticonnextdds-connector.js:43]: In py the logic is completely different for ARM archs

```
        if "Linux" in osname:
            # "Linux" can be ARMv7, ARMv8 or x64
            if "64" in bits:
                # ARMv8 can have the following strings returned by uname:
                # aarch64, aarch64_be, armv8b, armv8l
                # We want to match any of them
                if "aarch64" in machine or "armv8" in machine:
                    # ARMv8
                    directory = "linux-arm64"
                else:
                    # x64
                    directory = "linux-x64"
            elif "arm" in machine:
                # ARMv7
                directory = "linux-arm"
```

**Pull request merged**